packages/cozy-stack-client/src/PermissionCollection.js

```diff
@@ -105,33 +105,56 @@ class PermissionCollection extends DocumentCollection {
   }
 
   /**
-   * Create a share link
-   *
-   * @param {{_id, _type}} document - cozy document
-   * @param {object} options - options
-   * @param {string[]} options.verbs - explicit permissions to use
+   * Create a new set of permission on the stack
+   * @param {object} permissions - list of permissions
+   * @see https://docs.cozy.io/en/cozy-stack/permissions/#post-permissions
+   * @private
    */
-  async createSharingLink(document, options = {}) {
-    const { verbs } = options
+  async createPermissions(permissions) {
     const resp = await this.stackClient.fetchJSON(
       'POST',
       `/permissions?codes=email`,
       {
         data: {
           type: 'io.cozy.permissions',
           attributes: {
-            permissions: getPermissionsFor(
-              document,
-              true,
-              verbs ? { verbs } : {}
-            )
+            permissions: permissions
           }
         }
       }
     )
     return { data: normalizePermission(resp.data) }
   }
 
+  /**
+   * Create a share link
+   *
+   * @param {{_id, _type}} document - cozy document
+   * @param {object} options - options
+   * @param {string[]} options.verbs - explicit permissions to use
+   */
+  async createSharingLink(document, options = {}) {
+    const permissions = getPermissionsFor(document, true, options)
+    return this.createPermissions(permissions)
+  }
+
+  /**
+   * Create a share link from multiple documents
+   *
+   * @param {} documents - key/value
+   * where each value is either a document {_id, _type}
+   * or an array with a document and an option object
+   */
+  async createCompositeSharingLink(documents) {
+    const permissions = Object.keys(documents).map(key => {
+      const item = documents[key]
+      const document = Array.isArray(item) ? item[0] : item
+      const options = Array.isArray(item) ? item[1] : {}
+      return getPermissionsFor(document, true, { name: key, ...options })
+    })
+    return this.createPermissions(Object.assign({}, ...permissions))
+  }
```

Comment:
If I understand well, this is the responsibility of the callers to know that they should use a different method depending on whether they have several documents or only one document ?

If so, I think it would be ergonomic if `createSharingLink` could also accept an array of documents and call createCompositeSharingLink by itself.

- Only one method to know for the caller
- External API surface reduced

Reply:
I agree with you. I think having only one exposed method to create link is better.

Reply:
I wrote another function because the prototype was really different.

It's not just an array of document, otherwise I would agree with you.

It's a key/value of documents with their own options, which differ significantly from the first function. I don't feel confortable with having the same name with very different parameters.

`createSharingLink(document)`
`createSharingLink(document, { verbs: ['ALL'] })`
`createSharingLink({
   name1: [ document, { verbs: ['ALL'] } ],
   name2: [ document, { verbs: ['ALL'] } ],
})`

Reply:
s/prototype/signature I guess.

AMHA in this case we should deprecate the first API that proves itself not very flexible.

I would be comfortable having different parameters. But I understand your reason. Your call ;)